### ISSUE TITLE
Removed Option to Show Badge Count of Muted Chats

### ISSUE THREAD

**Issue by christophtrybek** [TelegramMessenger/Telegram-iOS#296]
The option to decide wether the user wants to receive updates to the badge count from muted chats was removed in an update.

In a previous version it was possible to change this setting in: Settings -> Notifications and Sounds -> Badge Counter

This functionality is still available in the macOS Application.

**Comment by ekumanov** [comment]
I also need this option restored. Why was it removed in the first place? I come from Skype and I'm very used to having muted chats with updated badge counter, so that I still know there are unread messages.

**Comment by ali-fareed** [comment]
Muted chats don't generate push notifications. At the moment the app is able to update the badges only when receiving a notification. The option may be restored in the future.

**Comment by ekumanov** [comment]
> Muted chats don't generate push notifications. At the moment the app is able to update the badges only when receiving a notification. The option may be restored in the future.

Is this a technical limitation? Can it be overcome? I've seen this working on Skype for iPhone, so I assume it's possible. I'm a Java engineer and not Swift but I can try to contribute if the reason is lack of enough resources?

**Comment by ali-fareed** [comment]
It is not impossible to implement. Unfortunately, a correct implementation would involve rewriting the data synchronization code completely.

**Comment by ekumanov** [comment]
I imagine that this option is the same as having full-blown (non-muted) mode for the group, however when the application receives its notification, it would only request the iOS to update only the badge without showing a banner/sound, etc. In other words, you won't need to rewrite the entire "muted" mode, but will instead just alter the "regular" mode depending on the option.

**Comment by ali-fareed** [comment]
It could be done, but it wastes too much battery already.

**Comment by ekumanov** [comment]
> It could be done, but it wastes too much battery already.

Yes, I can imagine that's true but it's no different than unmuting the chat which is what many people would do for small groups (a family/friend group). The use-case for muting in my case is not to save battery but to temporarily disable distractions while busy but not completely hide the chat. Anyway, I'd be glad if it will be easy to add this option, which won't be part of the muted mode but will be part of the regular mode and bear the meaning of "update only badge icon and suppress other notifications".
